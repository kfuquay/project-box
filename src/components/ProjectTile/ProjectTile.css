.tile {
    background: none;
    display: flex;
    flex-direction: column;
    border: 0;
    border-radius: 10px;
    margin: 0px 20px 40px 0;
    text-decoration: none;
    color: whitesmoke;
}

.tile:hover {
    cursor: pointer;
}

.tile-title {
    padding: 10px;
    flex: 1;
    max-width: 275px;
    background: #333;
    color: whitesmoke;
    margin: 0;
    border-radius: 0 0 10px 10px;
    text-align: center;
    font-size: 1em;
    letter-spacing: .09em;
    text-transform: uppercase;
}

.tile-img {
    background: darkviolet;
    height: 250px;
    width: 275px;
    border-radius: 10px 10px 0 0;
    background-size: cover;
    text-align: center;
    margin: 0;
    /* background-position: 50% 50%; */
}