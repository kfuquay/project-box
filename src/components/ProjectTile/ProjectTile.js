import React from "react";
import "./ProjectTile.css";
import { Link } from "react-router-dom";

function ProjectTile(props) {
  return (
    <Link to={`/project/${props.project.id}`} className="tile">
      <div className="tile-container">
        <p className="tile-img">{props.project.title.charAt(0).toUpperCase()}</p>
      </div>
      <h3 className="tile-title">{props.project.title}</h3>
    </Link>
  );
}
export default ProjectTile;
