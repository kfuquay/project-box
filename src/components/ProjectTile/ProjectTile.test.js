import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProjectTile from './ProjectTile';

describe(`ProjectTile Component`, () => {
  const project = {id: 1, img: 'blahs',}

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProjectTile />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders empty given no project', () => {
    const wrapper = shallow(<ProjectTile />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly given project prop', () => {
    const wrapper = shallow(<ProjectTile props={project} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})