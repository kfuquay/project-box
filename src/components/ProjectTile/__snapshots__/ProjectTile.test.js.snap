// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTile Component renders correctly given project prop 1`] = `
<Link
  className="tile"
  to="/project/1"
>
  <img
    alt=""
    className="tile-img"
    src="blahs"
  />
  <h3
    className="tile-title"
  />
</Link>
`;
